### PR TITLE
Add SSL Certificate v2 resource and data source for cluster management

### LINKED ISSUE #1019: Feature Request: SSL Certificate Management for Clusters
## Is your feature request related to a problem? Please describe.

Currently, there is no Terraform resource or data source to manage SSL certificates for Nutanix clusters. Users who need to:
- Update SSL certificates for clusters
- Regenerate self-signed certificates
- Read SSL certificate details

must do so manually through the Prism Central UI or via direct API calls. This creates several problems:
- No infrastructure-as-code approach for certificate management
- Manual processes are error-prone and time-consuming
- Difficult to automate certificate rotation and updates
- No way to track certificate changes in Terraform state
- Inconsistent certificate management across different environments

## Describe the solution you'd like

I would like to add Terraform resources and data sources for managing SSL certificates on Nutanix clusters:

### Resource: `nutanix_ssl_certificate_v2`
- **Import/Update SSL Certificates**: Allow users to import or update SSL certificates by providing:
  - Private key (Base64-encoded)
  - Public certificate (Base64-encoded)
  - CA chain (Base64-encoded)
  - Passphrase (optional)
  - Private key algorithm (RSA_2048, RSA_4096, ECDSA_256/384/521, etc.)

- **Regenerate Self-Signed Certificates**: Support regenerating self-signed certificates by specifying only the `private_key_algorithm` (RSA_2048 supported for regeneration)

- **Task Management**: Properly handle async operations with task polling to ensure certificate updates complete successfully

### Data Source: `data.nutanix_ssl_certificate_v2`
- Read SSL certificate details from a cluster
- Return all certificate information including public certificate, private key algorithm, CA chain, etc.
- Handle retries for transient API availability issues

### Key Features:
1. Support for multiple private key algorithms:
   - RSA_2048, RSA_4096
   - ECDSA_256, ECDSA_384, ECDSA_521
   - JKS, PKCS12, KRB_KEYTAB, RSA_PUBLIC

2. Base64 encoding/decoding for certificate data
3. Proper ETag handling for concurrent updates
4. Comprehensive error handling and retry logic
5. Full Terraform lifecycle support (create, read, update, delete)

## Describe alternatives you've considered

1. **Manual Management via UI**: 
   - Pros: Simple for one-off updates
   - Cons: Not scalable, error-prone, not version-controlled, can't be automated

2. **Direct API Calls via Scripts**:
   - Pros: Can be automated
   - Cons: No state management, requires custom scripts, no integration with Terraform workflows

3. **Using Existing Provider Resources**:
   - Pros: Would leverage existing infrastructure
   - Cons: No dedicated SSL certificate resources exist currently

4. **Third-party Tools**:
   - Pros: Might have certificate management features
   - Cons: Additional tooling, potential licensing costs, less integration with Nutanix ecosystem

The proposed solution (Terraform resources) is the best approach because:
- Integrates seamlessly with existing Terraform workflows
- Provides state management and change tracking
- Enables infrastructure-as-code practices
- Supports automation and CI/CD pipelines
- Maintains consistency across environments

### API Reference
- Nutanix Prism Central v4.1 SSL Certificate API
- Endpoint: `PUT /api/clustermgmt/v4.1/config/clusters/{clusterExtId}/ssl-certificate`
- Endpoint: `GET /api/clustermgmt/v4.1/config/clusters/{clusterExtId}/ssl-certificate`
- Documentation: https://developers.nutanix.com/api-reference?namespace=clustermgmt&version=v4.1#tag/SSLCertificate

### Use Cases
1. **Certificate Rotation**: Automate periodic SSL certificate updates
2. **Environment Setup**: Include certificate configuration in Terraform modules for new cluster deployments
3. **Compliance**: Track and manage certificate changes for audit purposes
4. **Disaster Recovery**: Restore certificate configurations as part of infrastructure recovery
5. **Multi-Environment Management**: Maintain consistent certificate policies across dev/staging/prod

### Example Usage (Proposed)

```hcl
# Regenerate self-signed certificate
resource "nutanix_ssl_certificate_v2" "cluster_cert" {
  cluster_ext_id        = "000643c6-8649-031f-6d6e-5254001006c3"
  private_key_algorithm = "RSA_2048"
}

# Import/update certificate with full chain
resource "nutanix_ssl_certificate_v2" "cluster_cert" {
  cluster_ext_id        = "000643c6-8649-031f-6d6e-5254001006c3"
  passphrase            = var.cert_passphrase
  private_key           = base64encode(file("private_key.pem"))
  public_certificate    = base64encode(file("certificate.pem"))
  ca_chain             = base64encode(file("ca_chain.pem"))
  private_key_algorithm = "RSA_2048"
}

# Read certificate details
data "nutanix_ssl_certificate_v2" "cluster_cert" {
  cluster_ext_id = "000643c6-8649-031f-6d6e-5254001006c3"
}
```

### Implementation Considerations
- Certificate updates are async operations that return task references
- Need to poll task status until completion
- API may be temporarily unavailable immediately after certificate regeneration
- ETag headers required for update operations to handle concurrent modifications
- Base64 encoding required for certificate data in API payloads

### Testing Requirements
- Unit tests for resource CRUD operations
- Integration tests with actual cluster
- Test certificate regeneration flow
- Test certificate import/update flow
- Test data source read operations
- Handle edge cases (missing certificates, API errors, etc.)

### PR DESCRIPTION
**Pull request by Haroon-Dweikat-Ntx**
## Description

This PR adds support for managing SSL certificates for Nutanix clusters through Terraform, including the ability to import/update certificates and regenerate self-signed certificates.

### New Resources and Data Sources
- **Resource**: `nutanix_ssl_certificate_v2` - Manage SSL certificates for clusters
  - Import/update SSL certificates with Base64-encoded certificate data
  - Regenerate self-signed certificates (RSA_2048 algorithm supported)
  - Support for passphrase, private key, public certificate, CA chain, and algorithm selection
  
- **Data Source**: `data.nutanix_ssl_certificate_v2` - Read SSL certificate details
  - Added retry logic with exponential backoff to handle temporary API unavailability
  - Supports reading certificate details after regeneration

### Features
- ✅ Import/update SSL certificates with full certificate chain
- ✅ Regenerate self-signed certificates (RSA_2048 only)
- ✅ Support for multiple private key algorithms (RSA_2048, RSA_4096, ECDSA_256/384/521, JKS, PKCS12, KRB_KEYTAB, RSA_PUBLIC)
- ✅ Task polling for certificate update operations
- ✅ Retry logic for data source reads to handle transient API issues
- ✅ Comprehensive test coverage

### Documentation
- Added resource documentation (`website/docs/r/ssl_certificate_v2.html.markdown`)
- Added data source documentation (`website/docs/d/ssl_certificate_v2.html.markdown`)
- Created example Terraform configurations (`examples/ssl_certificate_v2/`)

### Testing
- Unit tests for resource creation, update, and read operations
- Integration tests for certificate regeneration
- Tests include proper wait times and dependencies for certificate availability

### Technical Details
- Uses Nutanix Prism Central v4.1 SSL Certificate API
- Implements proper ETag handling for concurrent updates
- Task-based operations with state polling
- Base64 encoding/decoding for certificate data
- Error handling and retry mechanisms

## API Reference
- [Nutanix SSL Certificate v4 API](https://developers.nutanix.com/api-reference?namespace=clustermgmt&version=v4.1#tag/SSLCertificate/operation/updateSSLCertificate)

## Example Usage

# Regenerate self-signed certificate
resource "nutanix_ssl_certificate_v2" "example" {
  cluster_ext_id        = "cluster-uuid"
  private_key_algorithm = "RSA_2048"
}

# Import/update certificate
resource "nutanix_ssl_certificate_v2" "example" {
  cluster_ext_id        = "cluster-uuid"
  passphrase            = "password"
  private_key           = base64encode(var.private_key)
  public_certificate    = base64encode(var.public_certificate)
  ca_chain             = base64encode(var.ca_chain)
  private_key_algorithm = "RSA_2048"
}

# Read certificate details
data "nutanix_ssl_certificate_v2" "example" {
  cluster_ext_id = "cluster-uuid"
}## Testing
- [x] Unit tests pass
- [x] Integration tests pass
- [x] Documentation updated
- [x] Examples provided

## Related Issues
Closes #[issue-number]

### CODE REVIEW

File: examples/ssl_certificate_v2/main.tf [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Note - Replace appropriate values of variables in terraform.tfvars file as per setup
+
+terraform {
+  required_providers {
+    nutanix = {
+      source  = "nutanix/nutanix"
+      version = "2.0.0"
```
Comment (abhinavbansal29) [inline examples/ssl_certificate_v2/main.tf]:
```suggestion
      version = "2.4.0"
```
